datafusion/physical-expr/src/expressions/get_indexed_field.rs

```diff
@@ -105,9 +105,51 @@ impl PhysicalExpr for GetIndexedFieldExpr {
                 }
                 (dt, key) => Err(DataFusionError::NotImplemented(format!("get indexed field is only possible on lists with int64 indexes. Tried {} with {} index", dt, key))),
             },
-            ColumnarValue::Scalar(_) => Err(DataFusionError::NotImplemented(
-                "field access is not yet implemented for scalar values".to_string(),
-            )),
+            ColumnarValue::Scalar(scalar) => match (scalar.get_datatype(), &self.key) {
+                (DataType::List(v), ScalarValue::Int64(Some(i))) => {
+                    let wrapper = scalar.to_array();
```

Comment:
Should I implement a new method that will convert Vec<SQLExpr> (which ScalarValue holds) for List instead of using the current API and temporarily ArrayRef? For example: `as_array_list()`?

Thanks

cC @alamb

Reply:
I wonder if initially you might be able to use `ScalarValue::to_array_of_size` to convert the scalar argument into an `ArrayRef` and then use the same code as above:

Another alternative might be to use the take kernel with something like

Untested:

```rust
ColumnarValue::Scalar(scalar) =>  {
  let indicies = self.key.to_array_of_size(array.len());
  let indicies = arrow::compute::cast(&indicies, DataType::Int32);
  let values = arrow::compute::take(indices, array)
}
```

Though I see the code above to handle non scalars is a bit more involved so perhaps `take` doesn't work -- though there is code for lists here: https://github.com/apache/arrow-rs/blob/master/arrow/src/compute/kernels/take.rs#L222-L228

Reply:
`to_array` uses `to_array_of_size` under the hood, but I cannot use the code above, because it returns an `ColumnarValue::Array`, but in the case with `ColumnarValue::Scalar` we need to return `ColumnarValue::Scalar`.

I did another draft in https://github.com/apache/arrow-datafusion/pull/2196/commits/af8c77fcb947137a3d173ead288a2d7067c1be7d WDT?

Thanks

Reply:
@alamb Sorry for the ping, just a reminder that I am still waiting for advice on how to solve it. Thank you!

Reply:
sorry @ovr  -- taking a look